src/pages/workspaces/workspace/Dashboard.js

```diff
@@ -347,6 +348,34 @@ const WorkspaceDashboard = _.flow(
     }
   }
 
+  const loadAzureStorage = withErrorReporting('Error loading Azure storage information.', async () => {
+    try {
+      const { location, sas } = await Ajax(signal).AzureStorage.details(workspaceId)
+      setAzureStorage({ storageContainerUrl: _.head(_.split('?', sas.url)), storageLocation: location, sas })
+    } catch (error) {
+      // We expect to get a transient error while the workspace is cloning. We will improve
+      // the handling of this with WOR-534 so that we correctly differentiate between the
+      // expected transient error and a workspace that is truly missing a storage container.
+      console.log(`Error thrown by AzureStorage.details: ${error}`) // eslint-disable-line no-console
+    }
+  })
+
+  useEffect(() => {
+    if (isAzureWorkspace(workspace)) {
+      if (!storageContainerUrl && !interval.current) {
+        interval.current = setInterval(loadAzureStorage, 5000)
+      } else if (!!storageContainerUrl && interval.current) {
+        clearInterval(interval.current)
+        interval.current = undefined
+      }
+    }
+
+    return () => {
+      clearInterval(interval.current)
+      interval.current = undefined
+    }
+  }, [loadAzureStorage, workspace, storageContainerUrl])
```

Comment:
Since `loadAzureStorage` is created in this function, having it as a dependency here will cause the effect to be re-run on every render. To avoid this, `loadAzureStorage` could be wrapped with `useCallback` or `useMemo`.

Reply:
Thanks, I added useCallback.